### ISSUE TITLE
Dynamic import detection producing false positives when destructuring args

### ISSUE THREAD

**Issue by jp7837** [pzavolinsky/ts-unused-exports#123]
a.ts:
```
export const A = 1;
export const A_unused = 2;
```
b:ts:
```
import("./a").then({ A }=> {
  console.log(A);
});
export const B_unused = 0;
```
tsconfig.json:
```
{
	"compilerOptions": {
		"lib": ["ES2015", "DOM"]
	},
	"files": [
		"a.ts",
		"b.ts"
	]
}
```

**Expected**: A_unused, B_unused marked as unused
**Actual**: A, A_unused, B_unused marked as unused.

A should not be marked as unused since it's referenced in B.

**Comment by jp7837** [comment]
Very closely related to #117, but opening a new issue since it's slightly different.

**Comment by mrseanryan** [comment]
@jp7837 thanks for finding it

**Comment by mrseanryan** [comment]
Can also have renaming:

a.ts
```
export const A = 1;
export const A2 = 2;
export const A_unused = 3;
```

b.ts
```
  import('./a').then(({ A, A2: a2renamed }) => {
    console.log(A, a2renamed);
  });
```

**Comment by mrseanryan** [comment]
@jp7837 I think this one is fixed, in 6.1.1

**Comment by mrseanryan** [comment]
closing - unless something new comes up
